### PR TITLE
Service list

### PR DESCRIPTION
**Pull request by GarvinH**
Formats service inventory services into lists with description and service type

### CODE REVIEW

**Comment by GarvinH** [comment]
image

**Comment by ktw1016** [comment]
Hey Garvin, good start!! I was talking about this in Slack but I thought I should probably leave the comments here so it's more trackable.

1. Service types into something like `tag-badge`. I think service name should be more highlighted than service type. So it'd be nice if service type is a bit more subtle
2. I want it to be a bit more obvious that you can click to see more details on individual services. Possibly another button like `See this service` at the bottom right corner, something like this would be nice!
3. We'll need to be able to search through these services! Search input to filter by service name.
4. Speaking of filtering, filter by service types as well? :) I think this one can go into issues and be implemented later

**Comment by GarvinH** [comment]
image

image

**Comment by GarvinH** [comment]
I need to make "see this service" in french too

File: client/src/panels/panel_declarations/services/provided_services_list.js [hunk 1]
```diff
@@ -6,25 +6,80 @@ import {
...
+    super(props);
+    this.state = { query: "" };
+  }
+
+  render() {
+    const { panel_args } = this.props;
+
+    const filtered_services = _.filter(panel_args.services, (service) => {
+      if (
+        _.includes(service.name.toLowerCase(), this.state.query.toLowerCase())
```
Comment (ktw1016) [inline client/src/panels/panel_declarations/services/provided_services_list.js:22]: I personally prefer to declare a variable if object needs to be referenced multiple times like
`const { query } = this.state` or `const { services, subject } = panel_args`. This is optional personal preference though

I think variable name `query` can be a bit more specific also, like `service_query`?

Comment (ktw1016) [reply client/src/panels/panel_declarations/services/provided_services_list.js:22]: Also it'd be a bit cleaner to my taste without any if statements
`_.filter(services, service => _.includes(name) || _.includes(type)`

File: client/src/panels/panel_declarations/services/provided_services_list.js [hunk 2]
```diff
@@ -6,25 +6,80 @@ import {
...
+            subject_name: panel_args.subject.name,
+            num_of_services: panel_args.services.length,
+          }}
+        />
+        <input
+          aria-label={text_maker("explorer_search_is_optional")}
+          className="form-control input-lg"
+          type="text"
+          style={{ width: "100%", marginBottom: "1rem" }}
+          placeholder={text_maker("filter_results")}
```
Comment (ktw1016) [inline client/src/panels/panel_declarations/services/provided_services_list.js:51]: I think placeholder text can be a bit more specific, like `Filter services by name or type`?

**Comment by GarvinH** [comment]
image

French service type has bad input

File: client/src/panels/panel_declarations/services/provided_services_list.js [hunk 3]
```diff
@@ -6,25 +6,80 @@ import {
...
+                <div
+                  style={{
+                    display: "flex",
+                    fontSize: "14px",
+                    justifyContent: "space-between",
+                  }}
+                >
+                  <div className="tag-badge">{service_type}</div>
+                  <a href={`#dept/${org_id}/service-panels/${id}`}>
+                    See this service
```
Comment (ktw1016) [inline client/src/panels/panel_declarations/services/provided_services_list.js]: Should go into yaml file!

File: client/src/panels/panel_declarations/services/provided_services_list.js [hunk 4]
```diff
@@ -6,25 +6,80 @@ import {
...
+                <p>{description}</p>
+                <div
+                  style={{
+                    display: "flex",
+                    fontSize: "14px",
+                    justifyContent: "space-between",
+                  }}
+                >
+                  <div className="tag-badge">{service_type}</div>
+                  <a href={`#dept/${org_id}/service-panels/${id}`}>
```
Comment (ktw1016) [inline client/src/panels/panel_declarations/services/provided_services_list.js:70]: You should try styling this and see how it looks, I was originally thinking of styling it with something like `btn-ib-primary`

File: client/src/panels/panel_declarations/services/provided_services_list.js [hunk 5]
```diff
@@ -6,25 +6,80 @@ import {
...
+
+    return (
+      <div>
+        <TM
+          k="list_of_provided_services_desc"
+          args={{
+            subject_name: panel_args.subject.name,
+            num_of_services: panel_args.services.length,
+          }}
+        />
```
Comment (ktw1016) [inline client/src/panels/panel_declarations/services/provided_services_list.js:45]: A bit more separation between this text and input would be nice

**Comment by ktw1016** [comment]
> image
> 
> French service type has bad input

Ah this seems like data issue, don't mind these, I'll look into it with real data once I get that set up

File: client/src/panels/panel_declarations/services/provided_services_list.js [hunk 6]
```diff
@@ -1,30 +1,80 @@
...
-import {
-  create_text_maker_component,
-  InfographicPanel,
-  declare_panel,
-} from "../shared.js";
+import { InfographicPanel, declare_panel } from "../shared.js";
+import { FancyUL } from "../../../components";
 
-const { text_maker, TM } = create_text_maker_component(text);
+const text_maker = create_text_maker(text);
```
Comment (ktw1016) [inline client/src/panels/panel_declarations/services/provided_services_list.js:9]: any reasons why you needed to import `TM` separately?

File: client/src/panels/panel_declarations/services/provided_services_list.js [hunk 7]
```diff
@@ -1,30 +1,80 @@
...
+                <a href={`#dept/${org_id}/service-panels/${id}`}>{name}</a>
+                <p>{description}</p>
+                <div
+                  style={{
+                    display: "flex",
+                    fontSize: "14px",
+                    justifyContent: "space-between",
+                  }}
+                >
+                  <div className="tag-badge btn-ib-primary">{service_type}</div>
```
Comment (ktw1016) [inline client/src/panels/panel_declarations/services/provided_services_list.js:65]: ah I actually meant `btn-ib-primary` style for `See the service`. My bad on not being too clear on that haha

File: client/src/panels/panel_declarations/services/provided_services_list.js (hunk 6 above)
Comment (GarvinH) [reply client/src/panels/panel_declarations/services/provided_services_list.js:9]: Whoops. I guess I had a different idea of what the method did earlier haha

**Pull request closed without merging**